src/sqlancer/common/schema/AbstractTable.java

```diff
@@ -60,6 +60,15 @@ public C getRandomColumn() {
         return Randomly.fromList(columns);
     }
 
+    public C getRandomColumnWithFilter(String typeFilter) {
```

Comment:
I think this method should use a stream such as above (using the `filter` method), which means you can pass a `Predicate` that filters records. See https://www.javatpoint.com/java-8-stream-filter. Filtering based on the string representation of a type is not type-safe and thus error-prone.